### DISCUSSION TITLE
Cannot read property 'replace' of null

### DISCUSSION THREAD

**Discussion by michalchecinski** [renovatebot/renovate#17282] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please select which platform you are using if self-hosting.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When running renovate the following error occurs:

```
WARN: Error updating branch(branch="renovate/npm")
{
  "err": {
    "message": "Cannot read property 'replace' of null",
    "stack": "TypeError: Cannot read property 'replace' of null\n    at getAdditionalFiles (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/manager/npm/post-update/index.js:470:44)\n    at async processBranch (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/update/branch/index.js:313:33)\n    at async writeUpdates (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/write.js:30:21)\n    at async update (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/extract-update.js:111:15)\n    at async Object.renovateRepository (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/index.js:44:25)\n    at async renovateRepository (/home/ubuntu/renovateapp/app/worker/index.js:275:26)\n    at async /home/ubuntu/renovateapp/app/worker/index.js:539:5"
  }
}
```
Neither the PR nor branch is created.

Here is config used:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "schedule:monthly",
    ":maintainLockFilesMonthly",
    ":preserveSemverRanges",
    ":rebaseStalePrs"
  ],
  "enabledManagers": [
    "npm"
  ],
  "packageRules": [
    {
        "matchManagers": ["npm"],
        "groupName": "npm",
        "groupSlug": "npm",
        "separateMajorMinor": false
    }
  ]
}
```

Here is repo link: https://github.com/michalchecinski/clients

### Relevant debug logs

<details><summary>Logs</summary>

```
WARN: Error updating branch(branch="renovate/npm")
{
  "err": {
    "message": "Cannot read property 'replace' of null",
    "stack": "TypeError: Cannot read property 'replace' of null\n    at getAdditionalFiles (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/manager/npm/post-update/index.js:470:44)\n    at async processBranch (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/update/branch/index.js:313:33)\n    at async writeUpdates (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/write.js:30:21)\n    at async update (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/extract-update.js:111:15)\n    at async Object.renovateRepository (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/index.js:44:25)\n    at async renovateRepository (/home/ubuntu/renovateapp/app/worker/index.js:275:26)\n    at async /home/ubuntu/renovateapp/app/worker/index.js:539:5"
  }
}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment]
`separateMajorMinor` is nearly always a bad idea, especially if combined with grouping. You essentially guarantee yourself of having breaking, unmergeable PRs.

Additionally schedules match any, meaning that if the time is after 8am OR less than 9am then it will run, which is pretty much any time of the day, which means your schedule isn't really achieving anything.

I'm not sure what caused the error you saw, but you can see I'm now getting a PR in this reproduction repo: https://github.com/renovate-reproductions/17261

**Comment by michalchecinski** [comment]
Hello, thank you for the quick response. Using `separateMajorMinor`, I wanted to achieve only 1 PR for all NPM updates, to not distinguish between major, minor, etc. updates. I assume that `"matchUpdateTypes": ["major", "minor", "patch"],` does that?

For the schedules, when testing the config, I didn't want to wait for the schedule, I wanted to have the config changes immediately on the dependency dashboard and in PR.

**Reply by rarkins** [reply]
I strongly recommend you reconsider these settings for reasons I wrote above. And if you ignore me, remember to come back and read what I wrote above in future if you’re not enjoying the results.

I am converting this issue to a discussion as I was unable to reproduce it and this config is not recommended
